### PR TITLE
WIP: Add initial ToJSON

### PR DESCRIPTION
**Pull request by christiansakai**
**Issue reference:**
https://github.com/onrock-eng/github-webhooks/issues/12

This is my first attempt to add `ToJSON` instances. I am not sure how to do it better with line 689

I am also curious, how does `FromJSON` parses the time format for `created_at`, `updated_at`, and `closed_at`. I looked through the file and the repl, it seems there's no instance `FromJSON UTCTime` anywhere.

### CODE REVIEW

**Comment by donkeybonks** [comment]
There should be instances for those (at least I think they seem to work fine in the smoke tests?).

**Comment by christiansakai** [comment]
I see, it could be somewhere, I'll just have to look it up. I just wanted to double check.

File: src/GitHub/Data/Webhooks/Payload.hs [hunk 1]
```diff
@@ -660,6 +667,32 @@ instance FromJSON HookIssue where
...
+    , "user"          .= whIssueUser hookIssue
+    , "labels"        .= whIssueLabels hookIssue
+    , "state"         .= whIssueState hookIssue
+    , "locked"        .= whIssueIsLocked hookIssue
+    , "assignee"      .= whIssueAssignee hookIssue
+    , "milestone"     .= whIssueMilestone hookIssue
+    , "comments"      .= whIssueCommentCount hookIssue
+    , "created_at"    .= whIssueCreatedAt hookIssue
+    , "updated_at"    .= whIssueUpdatedAt hookIssue
+    , "closed_at"     .= (case whIssueClosedAt hookIssue of
```
Comment (joncfoo) [inline src/GitHub/Data/Webhooks/Payload.hs]: Does it make sense to return an empty string if there is no value for `closed_at` - i.e. it is `Nothing`?

Instead, how about about letting it possibly be `null` via `fmap toJSON (whIssueClosedAt hookIssue)` ?

Comment (christiansakai) [reply src/GitHub/Data/Webhooks/Payload.hs]: Sounds good. I'll do that

**Comment by donkeybonks** [comment]
What's left to do here?

**Comment by donkeybonks** [comment]
I'm going to close this because it's reasonably old. Anyone attempting this can try and rebase it.

**Pull request closed without merging**